frontend/ui/uimanager.lua

```diff
@@ -617,6 +627,33 @@ function UIManager:runForever()
     self:run()
 end
 
+-- Kobo does not have an auto suspend function, so we implement it ourselves.
+function UIManager:_initAutoSuspend()
+    local sec = G_reader_settings:readSetting("auto_suspend_timeout_seconds")
+    if sec then
+        self.auto_suspend_sec = sec
+    else
+        self.auto_suspend_sec = 15 * 60
+    end
+    self.auto_suspend_action = function()
+        local now = util.getsec()
+        -- Do not repeat auto suspend procedure after suspend.
+        if self.last_action_sec + self.auto_suspend_sec <= now then
+            Device:onPowerEvent("Suspend")
+        else
+            self:scheduleIn(
+                self.last_action_sec + self.auto_suspend_sec - now,
+                self.auto_suspend_action)
+        end
+    end
+    self:_startAutoSuspend()
+end
+
+function UIManager:_startAutoSuspend()
+    self.last_action_sec = util.getsec()
```

Comment:
Sorry, I am not an expert of lua :) I will change koreader-base. But I did not see a util.getrtime function.